### PR TITLE
Make CRUD tests use QueryEngine.

### PR DESCRIPTION
**Pull request by BearLemma**
Prerequisite to https://github.com/chiselstrike/chiselstrike/pull/1281 
Makes CRUD tests use query engine to provide better end-to-end coverage. It was very helpful when developing the above-mentioned PR and is imo worth the additional complexity.

### CODE REVIEW

File: server/src/datastore/crud.rs [hunk 1]
```diff
@@ -288,128 +291,6 @@ mod tests {
...
-            let ops = query_str_to_ops(&base_type, &url(".age=10")).unwrap();
-            assert_eq!(
-                ops,
-                vec![QueryOp::Filter {
-                    expression: binary(&["age"], BinaryOp::Eq, (10.).into())
-                }]
-            );
-
-            let ops =
-                query_str_to_ops(&base_type, &url(".age~lte=10&.name~unlike=foo%25")).unwrap();
```
Comment (espindola) [inline server/src/datastore/crud.rs:346]: Looks like a test for ~unlike and ~lte was lost.

File: server/src/datastore/crud.rs [hunk 2]
```diff
@@ -288,128 +291,6 @@ mod tests {
...
-                        QueryOp::Skip { count: 7 },
-                        QueryOp::Take { count: 3 },
-                    ],
-                    "unexpected ops for query string '{}'",
-                    query_string
-                );
-            }
-        }
-        {
-            let raw_ops = vec!["limit=3", "offset=7", "sort=age", ".age~gte=10"];
```
Comment (espindola) [inline server/src/datastore/crud.rs:383]: Looks like a test for gte was lost.

File: server/src/datastore/crud.rs [hunk 3]
```diff
@@ -477,35 +358,149 @@ mod tests {
...
+        add_row(qe, &PERSON_TY, &steve).await;
+
+        let mut r = run_query_vec("Person", url(""), qe).await;
+        r.sort();
+        assert_eq!(r, vec!["Alan", "John", "Steve"]);
+
+        let r = run_query_vec("Person", url("limit=2"), qe).await;
+        assert_eq!(r.len(), 2);
+
+        let r = run_query_vec("Person", url("limit=2"), qe).await;
```
[the remark this answers is not in the document]
Comment (BearLemma) [reply server/src/datastore/crud.rs:420]: fixed

File: server/src/datastore/crud.rs (hunk 2 above)
Comment (BearLemma) [reply server/src/datastore/crud.rs:383]: https://github.com/chiselstrike/chiselstrike/pull/1280/files#diff-20558519d1818bea6999042f85ad4ee0babd4ee084e3964bbeee7752021b0ee1R462 It's still there, I just changed the constants to get some results.

File: server/src/datastore/crud.rs (hunk 1 above)
Comment (BearLemma) [reply server/src/datastore/crud.rs:346]: Fixed

**Review by espindola — APPROVED** [review]

**Pull request merged**